common/lib/api-client/middleware/process-response.js

```diff
@@ -0,0 +1,27 @@
+const findUnpopulatedResources = require('../../find-unpopulated-resources')
+
+const processResponseMiddleware = {
+  name: 'process-response',
+  req: function req(response) {
+    const { req = {} } = response
+
+    // TODO: populateResources probably better as preserveResourceReferences
```

Comment:
This seems like a good rename to apply as part of this PR :)

Reply:
Sorted 👍